tests/integration/apigateway/test_apigateway_api.py

```diff
@@ -1528,3 +1528,241 @@ def test_update_model(
                 patchOperations=patch_operations,
             )
         snapshot.match("update-model-empty-schema", e.value.response)
+
+
+class TestApiGatewayApiRequestValidator:
+    @pytest.mark.aws_validated
+    def test_create_request_validator(self, apigateway_client, apigw_create_rest_api, snapshot):
+        response = apigw_create_rest_api(
+            name=f"test-api-{short_uid()}",
+            description="my api",
+        )
+        snapshot.match("create-rest-api", response)
+        api_id = response["id"]
+
+        response = apigateway_client.create_request_validator(
+            restApiId=api_id, name=f"test-validator-{short_uid()}"
+        )
+        snapshot.match("create-request-validator", response)
+
+    @pytest.mark.aws_validated
+    def test_get_request_validator(self, apigateway_client, apigw_create_rest_api, snapshot):
+        response = apigw_create_rest_api(
+            name=f"test-api-{short_uid()}",
+            description="my api",
+        )
+        snapshot.match("create-rest-api", response)
+        api_id = response["id"]
+
+        response = apigateway_client.create_request_validator(
+            restApiId=api_id, name=f"test-validator-{short_uid()}"
+        )
+        snapshot.match("create-request-validator", response)
+        validator_id = response["id"]
+
+        response = apigateway_client.get_request_validator(
+            restApiId=api_id, requestValidatorId=validator_id
+        )
+        snapshot.match("get-request-validators", response)
```

Comment:
We could add this to `test_create_request_validator`, as it is basically the same test extended with only one operation? Same for `test_get_request_validators`.

I feel like we could condense some of the tests with the flow:
- create a validator
- get the validator
- get all validators
- delete the validator
- try getting the deleted validator
- get all validators
This would look a lot like `test_delete_request_validator`, which cover the same ground as `test_create_request_validator` and `test_get_request_validators`.
With snapshot, we can take advantage of the fact that we can setup the resource one and follow its lifecycle.
We have good examples of this flow in the new lambda tests (see here:)https://github.com/localstack/localstack/blob/814bbe8613c8e955ff32274c7857828139711ef5/tests/integration/awslambda/test_lambda_api.py#L73

Otherwise, it's impeccable and I like how you structured the tests, and it's very clear, thanks a lot for tackling this!